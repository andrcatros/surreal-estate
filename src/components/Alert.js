import React from "react";

import "../styles/Alert.css";

const Alert = ({ alert }) => {
  if (alert.isSuccess) {
    return (
      <div
        className="Alert"
        style={{ borderStyle: "solid", borderColor: "green" }}
      >
        {alert.message}!
      </div>
    );
  }
  return (
    <div className="Alert" style={{ borderStyle: "solid", borderColor: "red" }}>
      {alert.message}!
    </div>
  );
};

export default Alert;
