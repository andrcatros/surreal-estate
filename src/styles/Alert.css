.Alert {
  position: absolute;
  margin-left: -275px;
  left: 50%;
  background-color: white;
  width: 550px;
  height: 30px;
  font-size: 20px;
}
