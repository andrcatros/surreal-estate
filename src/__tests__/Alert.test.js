import React from "react";
import { render, cleanup } from "@testing-library/react";
import Alert from "../components/Alert";
import "@testing-library/jest-dom/extend-expect";

describe("renders Alert component", () => {
  afterEach(cleanup);

  it("renders error Alert component", () => {
    const { getByText } = render(
      <Alert alert={{ message: "Error", isSuccess: false }} />
    );
    expect(getByText(/Error/i).textContent).toBe("Error!");
  });

  it("renders success Alert component", () => {
    const { getByText } = render(
      <Alert alert={{ message: "Success", isSuccess: true }} />
    );
    expect(getByText(/Success/i).textContent).toBe("Success!");
  });
});
