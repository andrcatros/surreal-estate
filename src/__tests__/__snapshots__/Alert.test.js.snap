// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Alert component renders error Alert component 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm iHdWFy"
    color="red"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`renders Alert component renders success Alert component 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm hHJyom"
    color="green"
  >
    Success!
  </div>
</DocumentFragment>
`;
